libc/test/src/math/sub_same_type_test.cpp

```diff
@@ -0,0 +1,22 @@
+//===-- Unittests for fputil::generic::sub --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "SubTest.h"
+
+#include "src/__support/FPUtil/generic/add_sub.h"
+#include "src/__support/macros/properties/types.h"
+
+#define SUB_FUNC(T) (LIBC_NAMESPACE::fputil::generic::sub<T, T>)
+
+LIST_SUB_SAME_TYPE_TESTS(Double, double, double, SUB_FUNC(double))
+LIST_SUB_SAME_TYPE_TESTS(Float, float, float, SUB_FUNC(float))
+LIST_SUB_SAME_TYPE_TESTS(LongDouble, long double, long double,
```

Comment:
I added MPFR support for float128.  Can you try to add float128 here to see if it works?

Reply:
Done.